api/src/main/java/io/onetable/spi/sync/TableFormatSync.java

```diff
@@ -59,10 +58,10 @@ public static TableFormatSync getInstance() {
    * @param snapshot the snapshot to sync
    * @return the result of the sync process
    */
-  public Map<TableFormat, SyncResult> syncSnapshot(
+  public Map<Object, SyncResult> syncSnapshot(
```

Comment:
Can we make these `String` instead of object? Same comment applies for the other maps below.

Reply:
I'm coming to the same conclusion as I am working through the nightmare of the test failures due to this ambiguity. I was trying to not disrupt existing code and allow for both. I think I am going to continue to try and fix it as is then will go back through and replace with String.

Reply:
Ok, take note of my above advice. This process will be smoother if you can break this up into a couple parts.

Reply:
Yes, understood. This has mostly been a discovery effort so far.